server/pulp/server/webservices/views/repo_groups.py

```diff
@@ -0,0 +1,370 @@
+
+        collection = RepoGroupModel.get_collection()
+        cursor = collection.find({})
+        groups = []
+        for group in cursor:
+            group['_href'] = reverse('repo_group_resource', kwargs={'repo_group_id': group['id']})
+            groups.append(group)
+        return generate_json_response_with_pulp_encoder(groups)
+
+    @auth_required(authorization.CREATE)
+    @json_body_allow_empty
+    def post(self, request):
+        """
+        Create a repo group from the data passed in the body.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+
+        :return: Response containing a serialized dict of the new repo group
+        :rtype: django.http.HttpResponse
+        :raises pulp_exceptions.MissingValue: if required values are not passed into the body
+        :raises pulp_exceptions.InvalidValue: if invalid values are passed into the body
+        """
+        group_data = request.body_as_json
+        group_id = group_data.pop('id', None)
+        if group_id is None:
+            raise pulp_exceptions.MissingValue(['id'])
+        display_name = group_data.pop('display_name', None)
+        description = group_data.pop('description', None)
+        repo_ids = group_data.pop('repo_ids', None)
+        notes = group_data.pop('notes', None)
+        distributors = group_data.pop('distributors', None)
+        if group_data:
+            raise pulp_exceptions.InvalidValue(group_data.keys())
+        # Create the repo group
+        manager = managers_factory.repo_group_manager()
+        args = [group_id, display_name, description, repo_ids, notes]
+        kwargs = {'distributor_list': distributors}
+        group = manager.create_and_configure_repo_group(*args, **kwargs)
+        group['_href'] = reverse('repo_group_resource', kwargs={'repo_group_id': group['id']})
+        group['distributors'] = distributors or []
+        response = generate_json_response_with_pulp_encoder(group)
+        return generate_redirect_response(response, group['_href'])
+
+
+class RepoGroupResourceView(View):
+    """
+    View for a single repo group.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, repo_group_id):
+        """
+        Retrieve the specified repo group.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_group_id: id of repo group to return
+        :type  repo_group_id: str
+
+        :return: Response containing serialized dict of the specified group
+        :rtype: django.http.HttpResponse
+        :raises pulp_exceptions.MissingResource: if repo_group_id is not found
+        """
+        collection = RepoGroupModel.get_collection()
+        group = collection.find_one({'id': repo_group_id})
+        if group is None:
+            raise pulp_exceptions.MissingResource(repo_group=repo_group_id)
+        group['_href'] = request.get_full_path()
```

Comment:
Here, I want to return the same path that was just called. This can be rebuilt using `reverse()`, but that introduces possibility for error, particularly if the code changes at some time in the future. I figure that this is just simpler.